### DISCUSSION TITLE
Failing Quickstart: docker-compose --env-file osmt.env up --build

### DISCUSSION THREAD

**Discussion by profnormano** [wgu-opensource/osmt#91] [Asking for Help] [answered]
Can anyone offer insight as to why this is failing?   Volume has 4GB free space

I was at this point in the build:

**The Quickstart configuration is deployed with a single docker-compose command. These steps below are the general process. Follow the guidance in Environment files for Quickstart and Development Stacks for more details.

Create file named osmt-quickstart.env in the project root. This file will be ignored by git.
From the project root, run this command:
docker-compose --env-file osmt-quickstart.env up --build**

Step 13/33 : RUN wget https://storage.googleapis.com/shellcheck/shellcheck-v$SHELLCHECK_VERSION.linux.x86_64.tar.xz -O- |     tar xJvf - shellcheck-v$SHELLCHECK_VERSION/shellcheck          &&     mv shellcheck-v$SHELLCHECK_VERSION/shellcheck /bin             &&     rmdir shellcheck-v$SHELLCHECK_VERSION                          &&     touch /tmp/libc.so.6                                           &&     echo "alias shellcheck='LD_LIBRARY_PATH=/tmp /bin/shellcheck'" >> /etc/bash.bashrc
 ---> Running in e33fbf362aee
--2021-12-19 00:30:36--  https://storage.googleapis.com/shellcheck/shellcheck-v0.5.0.linux.x86_64.tar.xz
Resolving storage.googleapis.com (storage.googleapis.com)... 142.250.69.208, 142.250.217.80, 142.250.217.112, ...
Connecting to storage.googleapis.com (storage.googleapis.com)|142.250.69.208|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 568 [application/x-xz]
Saving to: 'STDOUT'

     0K                                                       100% 3.98M=0s

2021-12-19 00:30:36 (3.98 MB/s) - written to stdout [568/568]

shellcheck-v0.5.0/shellcheck
 ---> 2ec1ace46e6f
Step 14/33 : RUN LD_LIBRARY_PATH=/tmp shellcheck -V
 ---> Running in e551edd03a54
/bin/sh: error while loading shared libraries: /tmp/libc.so.6: file too short
1 error occurred:
        * Status: The command '/bin/bash -o pipefail -c LD_LIBRARY_PATH=/tmp shellcheck -V' returned a non-zero code: 127, Code: 127

**Comment by profnormano** [comment]
I see this notice from https://shellcheck.storage.googleapis.com/index.html

ShellCheck builds have moved!
These builds were deprecated in March 2020, and were removed the following August. Get them from GitHub Releases instead.

Back to ShellCheck on GitHub

**Comment by profnormano** [comment]
I found this:

scversion="stable" # or "v0.4.7", or "latest"
wget -qO- "https://github.com/koalaman/shellcheck/releases/download/${scversion?}/shellcheck-${scversion?}.linux.x86_64.tar.xz" | tar -xJv
cp "shellcheck-${scversion}/shellcheck" /usr/bin/
shellcheck --version

at https://github.com/koalaman/shellcheck#installing

**Reply by profnormano** [reply]
Got past the above issue by adding the lines suggested by the koalaman git hub

**Comment by JohnKallies** [comment]
@profnormano This is great insight, thank you. I'm just seeing this after a holiday break.

We refactored the Dockerfile to use pinned "osmt-base" and "osmt-build" CentOS/Java11/Maven images from a wguopensource registry in DockerHub (https://github.com/wgu-opensource/osmt/pull/84). It didn't make sense to rebuild everything every time, and known-good images reduce variables like what you are describing. I can't account for the shellcheck resource being removed in August 2020, but I'm not experiencing any Docker build issues using the osmt-base image. 

Internally, we are working on a release mgt workflow friendly with Maven/git/GitHub releases. That should be completed within days, and we'll tag a new release version then. The develop branch should have the Docker image changes I'm talking about. Can you please pull from develop and verify that this is still a concern?

Please let me know if your observations remain after pulling from develop, and we can move your workaround to an issue for action.

Thank you!

**Reply by profnormano** [reply]
Switched to Development Configuration

From Development Configuration / Development Stacks / run ./exportEnv.sh osmt-dev-stack.env

received this error:  

Downloaded from jcenter: https://jcenter.bintray.com/org/testcontainers/testcontainers/1.15.3/testcontainers-1.15.3.jar (12 MB at 2.0 MB/s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:39 min
[INFO] Finished at: 2021-12-29T15:08:39Z
[INFO] Final Memory: 29M/71M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project osmt-api: Could not resolve dependencies for project edu.wgu.osmt:osmt-api:jar:1.0.5-SNAPSHOT: Could not find artifact edu.wgu.osmt:osmt-ui:jar:1.0.5-SNAPSHOT in spring-repo (https://repo.spring.io/release) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException

**Comment by JohnKallies** [comment, marked as answer]
Ah. Currently, the API module has a dependency on the UI module's build artifacts. We can probably remove that, but that's how it is now.

Step 2 of that section says to `mvn clean install` from the project root. `install` will put the build artifacts in your local Maven Repo. Maven is going our to jcenter because it can't find your build artifact locally.

**Reply by JohnKallies** [reply]
Hi Norman @profnormano. Can I check on how this is going? Did `mvn clean install` resolve it?

**Comment by profnormano** [comment]
Thank-you for the follow-up. I’m still working it.  The “mvn clean install” seemed to hang.  Today I’ll delete the OSMT directory and start again.   Sent from Mail for Windows From: John KalliesSent: Thursday, December 30, 2021 10:34 AMTo: wgu-opensource/osmtCc: profnormano; MentionSubject: Re: [wgu-opensource/osmt] Failing Quickstart: docker-compose --env-file osmt.env up --build (Discussion #91) Hi Norman @profnormano. Can I check on how this is going? Did mvn clean install resolve it?—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you were mentioned.Message ID: ***@***.***>

**Reply by JohnKallies** [reply]
Fair enough. To confirm, with a Java 11 and a recent Maven, `mvn clean install` in the project root should build the project. The NodeJS/npm used by the Maven build are managed by a Maven plugin, and do not rely on your globally-installed NodeJS/npm.

You can build the ui and api modules independently with `mvn clean install -pl ui` or `mvn clean install -pl api`.

Those things being said, the build tools for the Docker build are containerized and do not rely on your local SDKs and runtimes (see https://github.com/wgu-opensource/osmt/discussions/99)

At a minimum, make sure you pull latest from develop. There are some useful dev-tooling improvements being added there.
